karapace/schema_reader.py

```diff
@@ -199,33 +175,62 @@ def get_schema_id(self, new_schema: TypedSchema) -> int:
 
     def close(self) -> None:
         self.log.info("Closing schema_reader")
-        self.running = False
+        self._stop.set()
 
     def run(self) -> None:
-        while self.running:
-            try:
-                if not self.admin_client:
-                    if self.init_admin_client() is False:
-                        continue
-                if not self.schema_topic:
-                    if self.create_schema_topic() is False:
-                        continue
-                if not self.consumer:
-                    self.init_consumer()
-                self.handle_messages()
-            except Exception as e:  # pylint: disable=broad-except
-                if self.stats:
+        with ExitStack() as stack:
+            while not self._stop.is_set() and self.admin_client is None:
+                try:
+                    self.admin_client = _create_admin_client_from_config(self.config)
+                    stack.enter_context(closing(self.admin_client))
+                except (NodeNotReadyError, NoBrokersAvailable, AssertionError):
+                    self.log.warning("[Admin Client] No Brokers available yet. Retrying")
+                    self._stop.wait(timeout=2.0)
```

Comment:
I would rather make a `2.0` configurable property.

Reply:
I thought about this, but what would be the benefit of making this configurable?

Reply:
This `2.0` and the other `5.0` second timeout could be defined as constants.
```suggestion
                    self._stop.wait(timeout=CLIENT_CREATION_TIMEOUT_SECONDS)
                    # And for the topic creation TOPIC_CREATION_TIMEOUT_SECONDS
```

Reply:
```
KAFKA_CLIENT_CREATION_TIMEOUT_SECONDS = 2.0
SCHEMA_TOPIC_CREATION_TIMEOUT_SECONDS = 5.0
```

Reply:
> I thought about this, but what would be the benefit of making this configurable?

If I understand correctly, this number is just guesstimate, right? So we could tune it a bit later

Reply:
> If I understand correctly, this number is just guesstimate, right? So we could tune it a bit later

I don't know, I'm just moving the code around, these constants were already there. I understand we can tune the value, but I'm not sure if we need to make that a configuration setting, I think we should make configuration only values that we think are need to be tuned in production.